### PR TITLE
Starting to make changes to support nls objects

### PR DESCRIPTION
**Pull request by femiguez**
I'm starting to implement changes to support nls objects. Some of the functions do not need to be modified. A summary:

1. find_formula - works
2. find_parameters - proposed changes by FEM
3. find_predictors - works
4. find_response - works
5. find_statistic - works
6. find_terms - does not work, but should it?
7. find_variables - works
8. get_call - works
9. get_data - FEM copy of get_data.gls
10. get_deviance - works
11. get_intercept - works, it returns NA as it should for nls
12. get_loglikelihood - works
13. get_parameters - works
14. get_predicted - works

I'm considering what to do about 'get_predicted_ci'. Function 'predict_nls' in package 'nlraa' can produce what 'get_predict_ci' needs, but it is not set up as generic. So I'm open to suggestions about the best way to provide that functionality. In the nlraa package I also have prediction for 'gls', 'gnls', 'lme' and 'nlme' objects.

### CODE REVIEW

File: R/get_data.R [hunk 1]
```diff
@@ -216,13 +216,11 @@ get_data.lqmm <- function(x, verbose = TRUE, ...) {
...
+get_data.nls <- get_data.gls
 
-
-
-
-
-
-
+#' @export
+get_data.gnls <- get_data.gls
```
[the remark this answers is not in the document]
Comment (femiguez) [reply R/get_data.R:223]: That is intentional. There are objects in package nlme of class 'gnls' and that function should work. In fact, as written, it should probably also work for objects of class 'nlme', but I have not tested it.

[the remark this answers is not in the document]
Comment (strengejacke) [reply R/get_data.R:223]: ok, got it! thanks!

**Comment by strengejacke** [comment]
`find_terms()` should work, I guess. Do you have a small example where it fails?

@DominiqueMakowski can we merge this before we support get_predicted and address that later?

**Comment by femiguez** [comment]
I understand the distinction between terms and variables in linear mixed models, but it is not clear to me that this distinction is meaningful for nonlinear (mixed) models. I think the best approach would be for find_terms to behave the same way find_variables behaves, unless we want to exclude parameters. These are a couple of examples of what find_terms returns at the moment.

```
DNase1 <- subset(DNase, Run == 1)
## using a selfStart model
fm1DNase1 <- nls(density ~ SSlogis(log(conc), Asym, xmid, scal), DNase1)
find_terms(fm1DNase1) ## Not what we want
find_variables(fm1DNase1) ## This is better IMO, although 'conc' is a variable and Asym, xmid and scal are parameters
``` 
Perhaps find_terms should just return 'conc' instead. In this case it might seem obvious that it should be log(conc), but in the next example is not.

```
fm2DNase1 <- nls(density ~ 1/(1 + exp((xmid - log(conc))/scal)),
                 data = DNase1,
                 start = list(xmid = 0, scal = 1),
                 algorithm = "plinear")
find_terms(fm2DNase1) ## again not what we would want
find_variables(fm2DNase1) ## conc is a variable and xmid and scal are parameters
```

**Comment by strengejacke** [comment]
sounds reasonable, and if we still want any changes to this behaviour, we can still add this later. Thanks for the PR! Feel free to add yourself to the DESCRIPTION (with another PR, I will merge this now), if you like.

**Pull request merged**